cranelift/codegen/src/isa/mod.rs

```diff
@@ -227,6 +227,9 @@ pub trait TargetIsa: fmt::Display + Send + Sync {
     /// Get the ISA-independent flags that were used to make this trait object.
     fn flags(&self) -> &settings::Flags;
 
+    /// Get the ISA-dependent [`MachineEnv`] for managing register allocation.
+    fn machine_env(&self) -> &regalloc2::MachineEnv;
```

Comment:
Doesn't this depend on the calling convention in some cases? Like for the spidermonkey integration which uses to reserve some registers.

Reply:
That was actually handled as an ISA flag; in principle which registers we allocate within a function body and which ABI we implement at caller and callee boundaries are orthogonal, and changing which registers we use is an optimization (so controlling this via compiler settings is adequate).